Framework/DataHandling/src/LoadILLDiffraction.cpp

```diff
@@ -0,0 +1,446 @@
+  data.load();
+
+  // read the scan data
+  NXData scanGroup = firstEntry.openNXData("data_scan/scanned_variables");
+  NXDouble scan = scanGroup.openDoubleData();
+  scan.load();
+
+  // read which variables are scanned
+  NXInt scanned = firstEntry.openNXInt(
+      "data_scan/scanned_variables/variables_names/scanned");
+  scanned.load();
+
+  // read what is going to be the axis
+  NXInt axis =
+      firstEntry.openNXInt("data_scan/scanned_variables/variables_names/axis");
+  axis.load();
+
+  // read the starting two theta
+  NXFloat twoTheta0 = firstEntry.openNXFloat("instrument/2theta/value");
+  twoTheta0.load();
+
+  // figure out the dimensions
+  m_numberDetectorsRead = static_cast<size_t>(data.dim1());
+  m_numberScanPoints = static_cast<size_t>(data.dim0());
+
+  // set which scanned variables are scanned, which should be the axis
+  for (size_t i = 0; i < m_scanVar.size(); ++i) {
+    m_scanVar[i].setAxis(axis[static_cast<int>(i)]);
+    m_scanVar[i].setScanned(scanned[static_cast<int>(i)]);
+  }
+
+  resolveScanType();
+
+  resolveInstrument();
+
+  initWorkspace();
+
+  fillDataScanMetaData(scan);
+
+  if (m_scanType == DetectorScan) {
+    fillMovingInstrumentScan(data, scan);
+  } else {
+    fillStaticInstrumentScan(data, scan, twoTheta0);
+  }
+
+  scanGroup.close();
+  dataGroup.close();
+  firstEntry.close();
+  dataRoot.close();
+}
+
+/**
+* Dumps the metadata from the whole file to SampleLogs
+*/
+void LoadILLDiffraction::loadMetaData() {
+
+  m_outWorkspace->mutableRun().addProperty("Facility", std::string("ILL"));
+
+  // Open NeXus file
+  NXhandle nxHandle;
+  NXstatus nxStat = NXopen(m_fileName.c_str(), NXACC_READ, &nxHandle);
+
+  if (nxStat != NX_ERROR) {
+    m_loadHelper.addNexusFieldsToWsRun(nxHandle, m_outWorkspace->mutableRun());
+    nxStat = NXclose(&nxHandle);
+  }
+}
+
+/**
+ * Fills the loaded data to the workspace when the detector
+ * is not moving during the run, but can be moved before
+ * @param data : detector data
+ * @param scan : scan data
+ * @param twoTheta0 : starting two theta
+ */
+void LoadILLDiffraction::fillStaticInstrumentScan(const NXUInt &data,
+                                                  const NXDouble &scan,
+                                                  const NXFloat &twoTheta0) {
+
+  std::vector<double> axis = getAxis(scan);
+  std::vector<double> monitor = getMonitor(scan);
+
+  // Assign monitor counts
+  m_outWorkspace->mutableX(0) = axis;
+  m_outWorkspace->mutableY(0) = monitor;
+  std::transform(monitor.begin(), monitor.end(),
+                 m_outWorkspace->mutableE(0).begin(), sqrt);
+
+  // Assign detector counts
+  size_t deadOffset = (m_numberDetectorsRead - m_numberDetectorsActual) / 2;
+  for (size_t i = 1; i <= m_numberDetectorsActual; ++i) {
+    for (size_t j = 0; j < m_numberScanPoints; ++j) {
+      unsigned int y =
+          data(static_cast<int>(j), static_cast<int>(i - 1 + deadOffset));
+      m_outWorkspace->mutableY(i)[j] = y;
```

Comment:
move `m_outWorkspace->mutableY(i)` to a variable outside the inner most for loop